Newsletters: add 336 (2025-01-10)

harding opened the pull request:
- [x] Lede, releases/RCs, topic entries (including last week's entries)
- [x] Merges @Gustavojfe 

Note: we'd normally include a PR Review Club section but it looks like there's no meeting scheduled for this month.  If I'm missing something, @stickies-v please feel free to add the section.  You're also welcome to add it to any later newsletter this month.  I'll check in on the Club website again next month.

stickies-v commented:
There is indeed no review club this month - we're kicking off again this year on Feb 5th!

jonatack commented on _posts/en/newsletters/2025-01-10-newsletter.md:
s/In/If/ ? 

```suggestion
  would apply to all transactions required to settle a contract.  If
```

jonatack commented on _posts/en/newsletters/2025-01-10-newsletter.md:
perhaps s/problems/issues/ to avoid conflating with "the problem" in the next sentence

LarryRuane replied:
Agree, or I'd prefer "drawbacks".

jonatack commented on _posts/en/newsletters/2025-01-10-newsletter.md:
s/an HTLCs/HTLCs/ ?

jonatack commented on _posts/en/newsletters/2025-01-10-newsletter.md:
IIUC, "lead" here should be "led" (past tense) or "leads" (present tense)

harding replied:
Thanks.  Don't know how I keep making that mistake.

jonatack commented on _posts/en/newsletters/2025-01-10-newsletter.md:
nit, the frequent use of "gets confirmed" seems clunky, maybe prefer one of "is confirmed" / "have a state confirmed" / "<party> confirms" etc.

harding replied:
Agree it's clunky and I'll try to fix these.  I don't think it's important here (which is why I'm changing it), but I habitually use that language because confirming something isn't any individual's choice (unless they have 100% hashrate); it's probalistically determined based on overall miner choices.  C.f., I can't add code to Bitcoin Core (because I don't have commit access) but I can _get_ code added to Bitcoin Core (by following the regular process that ends with maintainer merging).

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,175 @@
+- **Investigating mining pool behavior before fixing a Bitcoin Core bug:**
+  Abubakar Sadiq Ismail [posted][ismail double] to Delving Bitcoin about
+  a [bug][bitcoin core #21950] discovered in 2021 by Antoine Riard that
+  results in it reserving 2,000 vbytes in block templates for coinbase
```
```suggestion
  results in the block-builder reserving 2,000 vbytes in block templates for coinbase
```
or "block-building algorithm"

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,175 @@
+  could include approximately five additional small transactions if the
+  double reservation was eliminated.  However, that could lead to miners
+  who depended on the double reservation producing invalid blocks,
+  resulting in a large loss of income.  Ismail analyzed past blocks to
```
```suggestion
  resulting in a significant loss of income.  Ismail analyzed past blocks to
```

harding replied:
The block reward is typically worth $300,000 to $500,000 USD.  Maybe this is just an indicator that my stack isn't big enough, but I still think a half-mil qualifies as large.  :smile:

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,175 @@
+  double reservation was eliminated.  However, that could lead to miners
+  who depended on the double reservation producing invalid blocks,
+  resulting in a large loss of income.  Ismail analyzed past blocks to
+  attempt to determine which mining pools might be at risk.  He noted
```
```suggestion
  determine which mining pools might be at risk.  He noted
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,175 @@
+  attempt to determine which mining pools might be at risk.  He noted
+  Ocean.xyz and F2Pool are apparently using non-default settings; he also
+  found an unknown pool or miner creating large coinbase transactions
+  that could be a problem if the Bitcoin Core bug is fixed.  We ask
```
```suggestion
  that could be problematic if the Bitcoin Core bug is fixed.  We ask
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  can be confirmed onchain, but only the last state confirmed before a
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  deadline can spend the channel funds.  Usually, the parties to
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  to resolve, called the _2x delay problem_.  That means
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  blockchain, showing how it could reduce the data required
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  cat].  Brandon Black [described][black penalty] an
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  Roberts proposed a scheme that allows a channel participant to
  get only a single state confirmed.  If a later state gets confirmed, the
```
(I'm not including Jon's suggestion here, though I agree with it)

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  can complete a state that no other party can complete, giving
```

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  to negative EV [expected value] pretty easily," even if the defender
```
(the comma doesn't need to be part of the actual quote, even though it's within quotation marks)

harding replied:
Leaving as-is.  I think it scans better without the comma.

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
- [Bitcoin Core #28121][] adds a "reject-details" field in testmempoolaccept verbose response
```

harding commented:
Made edits for all feedback so far, thanks @jonatack and @LarryRuane !  Also added topic entries from last week's newsletter (sorry for the delay).

azuchi commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
[news284 deltas]: /en/newsletters/2024/01/10/#expiry-deltas
```

bitschmidty commented on _topics/en/threshold-signature.md:
```diff
@@ -58,6 +58,9 @@ optech_mentions:
   - title: "BIP proposed for creating FROST scriptless threshold signatures for Bitcoin"
     url: /en/newsletters/2024/08/09/#proposed-bip-for-scriptless-threshold-signatures
 
+  - title: "Updated ChillDKG draft for distribute key generation for FROST"
```
```suggestion
  - title: "Updated ChillDKG draft for distributed key generation for FROST"
```

bitschmidty commented on _topics/en/transitory-soft-forks.md:
```diff
@@ -0,0 +1,65 @@
+between advocates of [OP_CHECKTEMPLATEVERIFY][topic
+op_checktemplateverify] and those who either doubted its utility or
+preferred to wait for a better alternative.  Since then, the idea has
+also been proposed to derisk consensus changes designed to bug fixes and
```
```suggestion
also been proposed to derisk consensus changes designed to fix bugs and
```

ismaelsadeeq commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,182 @@
+  double reservation was eliminated.  However, that could lead to miners
+  who depended on the double reservation producing invalid blocks,
+  resulting in a large loss of income.  Ismail analyzed past blocks to
+  determine which mining pools might be at risk.  He noted
+  Ocean.xyz and F2Pool are apparently using non-default settings; he also
+  found an unknown pool or miner creating large coinbase transactions
+  that could lead to them losing money if the Bitcoin Core bug is fixed
+  and they don't change their settings.  We ask
+  anyone who knows the unknown miner to alert them to the situation.
```
The unknown miner is likely not affected. Based on my data, their block template was `1,887,437`, which suggests they may have settings similar to Ocean, i.e., using a lowered `-blockmaxweight` option.

Could add this suggestion below or something like it.

```suggestion
He also noted noting that Ocean.xyz, an unknown pool/miner were creating large coinbase transactions and F2Pool creating optimal block templates (not affected by the bug).
Abubakar noted that If the bug is fixed, any miner relying on the previous behavior could produce an invalid block.
As a result, the discussion is shifting toward fixing the bug by consolidating the reservation in one place while introducing a new startup option `-maxcoinbaseweight=8000`, which is backward compatible.

Normal miners can just start Bitcoin Core with the new `-maxcoinbaseweight=4000`, while pools that need more space like Ocean can start up normally and retain the previous behavior.
```

ismaelsadeeq commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```diff
@@ -0,0 +1,182 @@
+  a [bug][bitcoin core #21950] discovered in 2021 by Antoine Riard that
+  results in nodes reserving 2,000 vbytes in block templates for coinbase
+  transactions rather than the intended 1,000 vbytes.  Each template
+  could include approximately five additional small transactions if the
+  double reservation was eliminated.  However, that could lead to miners
```
nit:
The number 5 is relative to the size of the small transactions, maybe indicate?

harding replied:
Leaving as-is.  Anyone who cares can divide 1,000 vbytes by five 5 transactions to figure out that we're talking about roughly 200-vbyte transactions.  We add this detail to help people quickly understand the impact of this bug, which is (at worst) five 2-in-2-out transactions; providing them additional information that they can easily figure out for themselves if they care makes it less quick and less useful, IMO.

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  confirm only a single state.  If a later state is confirmed, the
```
I'm not sure if I need to justify this suggestion (I made a similar one elsewhere in this document, and in earlier newsletters), but just in case: This may be pedantic, maybe I'm overthinking it, but as written, a channel participant can do only one thing overall, like there's nothing else a channel participant can do. But that's not true, is it? A channel participant can, for example, not confirm any states, just not do anything.

I think the desired meaning is that "only" applies to the number of states; the participant can do lots of other stuff, but if he wants to confirm, it's limited to _one particular_ state. So in a way, "only" and "single" are redundant, but I that's perfectly normal and acceptable.

harding replied:
No need to justify (but thank you for doing so).  For small edits, I don't normally look around the document to see where else I should apply the same edit.  I do seem to have a problem with often putting "only" in the wrong place.

LarryRuane commented on _posts/en/newsletters/2025-01-10-newsletter.md:
```suggestion
  a channel would attempt to confirm only the latest state; however, if
```

harding commented:
Addressed all feedback, thanks @azuchi @ismaelsadeeq @bitschmidty @LarryRuane !  Added lede, updated releases/RCs, and added topic links.  Review @Gustavojfe section (great as always!).

Thanks everyone!

bitschmidty reviewed (approved):
ACK lede, new topics, fixups

ismaelsadeeq reviewed (approved):
LGTM
